rosbag2_transport/include/rosbag2_transport/recorder.hpp

```diff
@@ -172,8 +173,8 @@ class Recorder : public rclcpp::Node
 
   // Variables for event publishing
   rclcpp::Publisher<rosbag2_interfaces::msg::WriteSplitEvent>::SharedPtr split_event_pub_;
-  bool event_publisher_thread_should_exit_ = false;
-  bool write_split_has_occurred_ = false;
+  std::atomic<bool> event_publisher_thread_should_exit_ = false;
```

Comment:
To get around the ABI break - maybe there's a way to introduce some of this synchronization via a new non-virtual member function?

Reply:
Yeah, I am not closing this PR as would not be done. I want to try to backport it with some workarounds.